src/d3a/models/state.py

```diff
@@ -43,131 +46,249 @@
 # - If a device has no state, maybe it doesn't need its own appliance class either
 
 
-class PVState:
+class StateInterface(metaclass=ABCMeta):
+    """Interface containing methods that need to be defined by each State class."""
+
+    @abstractmethod
+    def get_state(self) -> Dict:
+        """Return the current state of the device."""
+        return {}
+
+    @abstractmethod
+    def restore_state(self, state_dict: Dict):
+        pass
+
+    @abstractmethod
+    def delete_past_state_values(self, time_slot):
+        pass
+
+    def __str__(self):
+        return self.__class__.__name__
+
+
+class ConsumptionState(StateInterface):
+    """State for devices that can consume energy."""
+
     def __init__(self):
+        super().__init__()
+        # Energy that the load wants to consume (given by the profile or live energy requirements)
+        self._desired_energy_Wh = {}
+        # Energy that the load needs to consume. It's reduced when new energy is bought
+        self._energy_requirement_Wh = {}
+        self._total_energy_demanded_Wh = 0
+
+    def get_state(self) -> Dict:
+        """Return the current state of the device. Extends super implementation."""
+        state = super().get_state()
+        consumption_state = {
+            "desired_energy_Wh": convert_pendulum_to_str_in_dict(self._desired_energy_Wh),
+            "total_energy_demanded_Wh": self._total_energy_demanded_Wh}
+        # The inherited state should not have keys with the same name (to avoid overwriting them)
+        conflicting_keys = state.keys() & consumption_state.keys()
+        assert not conflicting_keys, f"Conflicting state values found for {conflicting_keys}."
+
+        state.update(consumption_state)
+        return state
+
+    def restore_state(self, state_dict: Dict):
+        super().restore_state()
+
+        self._desired_energy_Wh.update(convert_str_to_pendulum_in_dict(
+            state_dict["desired_energy_Wh"]))
+        self._total_energy_demanded_Wh = state_dict["total_energy_demanded_Wh"]
+
+    def get_energy_requirement_Wh(self, time_slot, default_value=0.0):
+        if default_value is None:
+            return self._energy_requirement_Wh[time_slot]
```

Comment:
IMO the default value should always be fallen to, what if the interface gets a None and the key doesn't exist ?

Reply:
I agree. I had changed it for the other methods below but I missed this one.